shell/platform/darwin/macos/framework/Source/FlutterKeyboardManager.mm

```diff
@@ -57,7 +57,7 @@ @interface FlutterKeyboardManager ()
 /**
  * The text input plugin set by initialization.
  */
-@property(nonatomic) id<FlutterKeyboardViewDelegate> viewDelegate;
+@property(nonatomic, weak) id<FlutterKeyboardViewDelegate> viewDelegate;
```

Comment:
Why weak? It might be worth documenting who's guaranteeing ownership in this relationship.

Reply:
This `viewDelegate` is actually the FVC, and it must be weak because the FVC owns this `FlutterKeyboardManager`, and this cyclic ownership causes memory leak.

As I've documented in [FlutterViewController](https://github.com/flutter/engine/blob/main/shell/platform/darwin/macos/framework/Headers/FlutterViewController.h#L42), FVCs strongly reference FE, and FE weakly references FVCs. The user should hold the only strong reference to FVC (in practice, that strong reference is then moved to the NSWindow). If that  strong reference is no more, then the FVC is deallocated by the system, when removes itself from the FE. Therefore, this memory leak even has behavioral significance.